main.go

```diff
@@ -162,12 +170,35 @@ func main() {
 
 	// Detect Project ID
 	if *gcpProjectID == "" {
```

Comment:
I would have thought we can use https://pkg.go.dev/golang.org/x/oauth2/google?tab=doc#Credentials

This looks like it would work for any auth method no? It suggests it may not always be populated though.

Also, another option might be to skip this entirely and pass it in (we know it from the kustomize cluster properties)

Reply:
Note, it's typically `DefaultCredentials` that we access this by (rather than just `Credentials`)

https://github.com/golang/oauth2/blob/bf48bf16ab8d/google/default.go#L38

Reply:
we do need the `gcpProjectID` to call the scrape quota query. 

We do use the defaultClient credentials but we still need the project as a string so we now do

1. If passed as either a cli argument or env variable use that ( might want to monitor a separate project )
1. If not passed but credentials.json file is defined extract from that ( which is this PR )
1. If none of the above try to fetch from compute metadata 
1. Fail